### PR TITLE
Add a name property to error structures

### LINKED ISSUE #94: Add name to exceptions in models/index.ts
**Describe the bug**
Add name to exceptions in models/index.ts
Refs:
* PR which added name to error https://github.com/awslabs/smithy-typescript/pull/92
* CodeGen CI failure https://github.com/aws/aws-sdk-js-v3/pull/724#issuecomment-573847987

**To Reproduce**
Run CodeGen with latest smithy-typescript having fix in https://github.com/awslabs/smithy-typescript/pull/92

**Expected behavior**
CodeGen CI succeeds

### PR DESCRIPTION
**Pull request by kstich**
Fixes #94 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/StructureGenerator.java [hunk 1]
```diff
@@ -158,6 +158,7 @@ private void renderErrorStructure() {
 
         writer.openBlock("export interface $L extends $L {", symbol.getName(), extendsFrom);
         writer.write("__type: $S;", shape.getId().getName());
+        writer.write("name: $S;", shape.getId().getName());
```
Comment (mtdowling) [inline smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/StructureGenerator.java]: I thought we were just special casing exceptions. Why is this needed on all structures? Why would we need this *and* __type?

Comment (kstich) [reply smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/StructureGenerator.java]: This is in the code for exception structures: `renderErrorStructure`

Comment (mtdowling) [reply smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/StructureGenerator.java]: Ok. This needs to be addressed in a followup PR.

Comment (trivikr) [reply smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/StructureGenerator.java]: Issue created for removing __type https://github.com/awslabs/smithy-typescript/issues/96
